pkg/plugins/policies/meshtrafficpermission/plugin/v1alpha1/plugin.go

```diff
@@ -84,3 +88,46 @@ func (p plugin) Apply(rs *core_xds.ResourceSet, ctx xds_context.Context, proxy *
 	}
 	return nil
 }
+
+func (p plugin) configureEgress(rs *core_xds.ResourceSet, proxy *core_xds.Proxy) error {
+	listeners := policies_xds.GatherListeners(rs)
+	for _, resource := range proxy.ZoneEgressProxy.MeshResourcesList {
+		if !resource.Mesh.MTLSEnabled() {
+			log.V(1).Info("skip applying MeshTrafficPermission, MTLS is disabled",
+				"mesh", resource.Mesh.GetMeta().GetName())
+			continue
+		}
+		for _, es := range resource.ExternalServices {
+			meshName := resource.Mesh.GetMeta().GetName()
+			esName := es.Meta.GetName()
+			policies, ok := resource.Dynamic[esName]
+			if !ok {
+				continue
+			}
+			mtp, ok := policies[api.MeshTrafficPermissionType]
+			if !ok {
+				continue
+			}
+			l := listeners.Egress[core_rules.InboundListener{
+				Address: proxy.ZoneEgressProxy.ZoneEgressResource.Spec.Networking.Address,
+				Port:    proxy.ZoneEgressProxy.ZoneEgressResource.Spec.Networking.Port,
+			}]
+
+			for _, rule := range mtp.FromRules.Rules {
+				configurer := &v3.RBACConfigurer{
+					StatsName: l.Name,
+					Rules:     rule,
+					Mesh:      resource.Mesh.GetMeta().GetName(),
+				}
+				for _, filterChain := range l.FilterChains {
+					if filterChain.Name == fmt.Sprintf("%s_%s", esName, meshName) {
```

Comment:
Could you please create a method like we have in https://github.com/kumahq/kuma/blob/master/pkg/xds/envoy/names/resource_names.go